discovery/kubernetes/node.go

```diff
@@ -140,16 +141,31 @@ func nodeSourceFromName(name string) string {
 
 const (
 	nodeNameLabel        = metaLabelPrefix + "node_name"
+	nodeRoleLabel        = metaLabelPrefix + "node_role"
 	nodeLabelPrefix      = metaLabelPrefix + "node_label_"
 	nodeAnnotationPrefix = metaLabelPrefix + "node_annotation_"
 	nodeAddressPrefix    = metaLabelPrefix + "node_address_"
+
+	rolePrefix = "node-role.kubernetes.io/"
+	roleLabel  = "kubernetes.io/role"
 )
 
 func nodeLabels(n *apiv1.Node) model.LabelSet {
 	ls := make(model.LabelSet, len(n.Labels)+len(n.Annotations)+1)
 
 	ls[nodeNameLabel] = lv(n.Name)
 
+	for k, v := range n.Labels {
```

Comment:
We already have this under node_label as far as I can see.

Reply:
yes exactly, i change that

Reply:
My point is that this is already supported and available to relabelling, so your change isn't needed.

Reply:
yes when we use `kubernetes.io/role="worker"`, it's simple.
But with `node-role.kubernetes.io/worker=""`, how would you do with this syntax? I tried different things without success.

Reply:
Hmm, I suspect that you're hitting https://github.com/prometheus/prometheus/issues/3726.